File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/WandCoreMaterials.java
package me.alegian.thavma.impl.init.registries.deferred;

import me.alegian.thavma.impl.Thavma;
import me.alegian.thavma.impl.common.wand.WandCoreMaterial;
import me.alegian.thavma.impl.init.registries.deferred.util.DeferredWandCoreMaterial;
import me.alegian.thavma.impl.init.registries.deferred.util.T7DeferredRegister;

public class WandCoreMaterials {
  public static final T7DeferredRegister.WandCoreMaterials REGISTRAR = T7DeferredRegister.createWandCoreMaterials(Thavma.MODID);

  public static final DeferredWandCoreMaterial<WandCoreMaterial> WOOD = register("wood", 25);
  public static final DeferredWandCoreMaterial<WandCoreMaterial> GREATWOOD = register("greatwood", 50);
  public static final DeferredWandCoreMaterial<WandCoreMaterial> SILVERWOOD = register("silverwood", 100);

  private static DeferredWandCoreMaterial<WandCoreMaterial> register(String name, int capacity) {
    return REGISTRAR.registerWandCoreMaterial(name, () -> new WandCoreMaterial(capacity));
  }
}

File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/WandCoreMaterials.kt
package me.alegian.thavma.impl.init.registries.deferred

import me.alegian.thavma.impl.Thavma
import me.alegian.thavma.impl.common.wand.WandCoreMaterial
import me.alegian.thavma.impl.init.registries.T7Registries
import net.neoforged.neoforge.registries.DeferredRegister

object WandCoreMaterials {
    val REGISTRAR = DeferredRegister.create(T7Registries.WAND_CORE.key(), Thavma.MODID)

    val WOOD = REGISTRAR.register("wood") { -> WandCoreMaterial(25) }

    val GREATWOOD = REGISTRAR.register("greatwood") { -> WandCoreMaterial(50) }

    val SILVERWOOD = REGISTRAR.register("silverwood") { -> WandCoreMaterial(100) }
}
